go/vt/vtgate/engine/insert.go

```diff
@@ -386,267 +138,14 @@ func (ins *Insert) executeInsertQueries(
-		if err != nil {
-			return 0, err
-		}
-		// If no rows are returned, it's an internal error, and the code
-		// must panic, which will be caught and reported.
-		insertID, err = qr.Rows[0][0].ToCastInt64()
-		if err != nil {
-			return 0, err
-		}
-	}
-
-	// Fill the holes where no value was supplied.
-	cur := insertID
-	for i, v := range values {
-		if shouldGenerate(v) {
-			bindVars[SeqVarName+strconv.Itoa(i)] = sqltypes.Int64BindVariable(cur)
-			cur++
-		} else {
-			bindVars[SeqVarName+strconv.Itoa(i)] = sqltypes.ValueBindVariable(v)
-		}
-	}
-	return insertID, nil
-}
-
-// processGenerateFromRows generates new values using a sequence if necessary.
-// If no value was generated, it returns 0. Values are generated only
-// for cases where none are supplied.
-func (ins *Insert) processGenerateFromRows(
-	ctx context.Context,
-	vcursor VCursor,
-	rows []sqltypes.Row,
-) (insertID int64, err error) {
-	if ins.Generate == nil {
-		return 0, nil
-	}
-	var count int64
-	offset := ins.Generate.Offset
-	genColPresent := offset < len(rows[0])
-	if genColPresent {
-		for _, val := range rows {
-			if val[offset].IsNull() {
-				count++
-			}
-		}
-	} else {
-		count = int64(len(rows))
-	}
-
-	if count == 0 {
-		return 0, nil
-	}
-
-	// If generation is needed, generate the requested number of values (as one call).
-	rss, _, err := vcursor.ResolveDestinations(ctx, ins.Generate.Keyspace.Name, nil, []key.Destination{key.DestinationAnyShard{}})
-	if err != nil {
-		return 0, err
-	}
-	if len(rss) != 1 {
-		return 0, vterrors.Errorf(vtrpcpb.Code_INVALID_ARGUMENT, "auto sequence generation can happen through single shard only, it is getting routed to %d shards", len(rss))
-	}
-	bindVars := map[string]*querypb.BindVariable{"n": sqltypes.Int64BindVariable(count)}
-	qr, err := vcursor.ExecuteStandalone(ctx, ins, ins.Generate.Query, bindVars, rss[0])
-	if err != nil {
-		return 0, err
-	}
-	// If no rows are returned, it's an internal error, and the code
-	// must panic, which will be caught and reported.
-	insertID, err = qr.Rows[0][0].ToCastInt64()
-	if err != nil {
-		return 0, err
-	}
-
-	used := insertID
-	for idx, val := range rows {
-		if genColPresent {
-			if val[offset].IsNull() {
-				val[offset] = sqltypes.NewInt64(used)
-				used++
-			}
-		} else {
-			rows[idx] = append(val, sqltypes.NewInt64(used))
-			used++
-		}
-	}
-
-	return insertID, nil
-}
-
-// getInsertShardedRoute performs all the vindex related work
+// getInsertQueries performs all the vindex related work
 // and returns a map of shard to queries.
 // Using the primary vindex, it computes the target keyspace ids.
 // For owned vindexes, it creates entries.
 // For unowned vindexes with no input values, it reverse maps.
 // For unowned vindexes with values, it validates.
 // If it's an IGNORE or ON DUPLICATE key insert, it drops unroutable rows.
-func (ins *Insert) getInsertShardedRoute(
+func (ins *Insert) getInsertQueries(
```

Comment:
hard to see in this diff, but this method is too large and deeply nested to be easy to overview. I think it would benefit from extracting out methods